rest-client/impl/impl-base/src/main/java/org/jboss/arquillian/extension/rest/client/BaseRestEnricher.java

```diff
@@ -89,25 +86,25 @@ private void addHeaders(Map<String, String> headersMap, Header annotation)
         }
     }
 
-    protected boolean allInSameContext(List<Servlet> servlets)
-    {
-        Set<String> context = new HashSet<String>();
-        for (Servlet servlet : servlets) {
-            context.add(servlet.getContextRoot());
-        }
-        return context.size() == 1;
-    }
-
     protected abstract Object enrichByType(Class<?> clazz, Method method, ArquillianResteasyResource annotation, Consumes consumes, Produces produces);
 
-    // Currently no way to share @ArquillianResource URL (URLResourceProvider) logic internally, copied logic
+    // Currently no way to request value of URI, need to manually invoke the lookup
     protected URI getBaseURL()
     {
-        HTTPContext context = metaDataInst.get().getContext(HTTPContext.class);
-        if (allInSameContext(context.getServlets())) {
-            return context.getServlets().get(0).getBaseURI();
+        Collection<ResourceProvider> resourceProviders = loader.get().all(ResourceProvider.class);
+        for(ResourceProvider resourceProvider: resourceProviders)
+        {
+            if(resourceProvider.canProvide(URI.class))
```

Comment:
The same issue is in core today.  If i have multiple providers for URI, I simply get the first.  If you recall, I pinged you a while ago about disabling the built in one, which lead to this little hack

https://github.com/hammock-project/hammock/blob/master/test-arquillian/src/main/java/ws/ament/hammock/test/support/HammockArquillianExtension.java#L31